src/onefetch/info.rs

```diff
@@ -131,14 +132,41 @@ impl std::fmt::Display for Info {
                 let pad = " ".repeat(title.len());
                 let mut s = String::from("");
                 let languages: Vec<(String, f64)> = {
-                    let mut iter = self.languages.iter().map(|x| (format!("{}", x.0), x.1));
+                    let mut language_map: HashMap<String, f64> = HashMap::new();
+                    let mut deduped_languages = Vec::new();
+
+                    // This part dedupes the languages
+                    // I.e if there are two entries for "shell" then it adds them
+                    for language in self.languages.iter() {
+                        let name = format!("{}", language.0);
+
+                        if language_map.contains_key(&name) {
+                            let val = language_map[&name] + language.1;
+
+                            language_map.insert(name, val);
+                        } else {
+                            language_map.insert(name, language.1);
+                        }
+                    }
+
+                    for language in language_map.iter() {
+                        deduped_languages.push((language.0, *language.1))
+                    }
+
+                    let mut deduped_iter =
+                        deduped_languages.iter().map(|x| (format!("{}", x.0), x.1));
+
+                    std::println!("{:?}", language_map.iter());
+
                     if self.languages.len() > 6 {
-                        let mut languages = iter.by_ref().take(6).collect::<Vec<_>>();
-                        let other_sum = iter.fold(0.0, |acc, x| acc + x.1);
+                        let mut languages = deduped_iter.by_ref().take(6).collect::<Vec<_>>();
+                        let other_sum = deduped_iter.fold(0.0, |acc, x| acc + x.1);
                         languages.push(("Other".to_owned(), other_sum));
                         languages
                     } else {
-                        iter.collect()
+                        // HELP WANTED can't sort by percentage
+                        let languages = deduped_iter.collect::<Vec<_>>();
+                        languages
                     }
```

Comment:
I can't seem to figure out how to sort the stats by percentage. I've never touched Rust so I have no idea what to do